java/src/jmri/managers/AbstractStringIOManager.java

```diff
@@ -32,6 +34,67 @@ public int getXMLOrder() {
         return Manager.STRINGIOS;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    @Nonnull
+    public StringIO provideStringIO(@Nonnull String name) {
+        var t = getStringIO(name);
+        if (t == null) {
+            t = newStringIO(makeSystemName(name), null);
+        }
+        return t;
+    }
+
+    private StringIO getStringIO(@Nonnull String name) {
+        var t = getByUserName(name);
+        if (t != null) {
+            return t;
+        }
+        return getBySystemName(name);
+    }
+
+    /**
+     * Create a New StringIO.
+     * {@inheritDoc}
+     */
+    @Override
+    @Nonnull
+    final public StringIO newStringIO(@Nonnull String systemName, String userName) throws IllegalArgumentException {
+        log.debug(" newStringIO(\"{}\", \"{}\")", systemName, (userName == null ? "null" : userName));
+        Objects.requireNonNull(systemName, "SystemName cannot be null. UserName was "
+                + (userName == null ? "null" : userName));  // NOI18N
+        systemName = validateSystemNameFormat(systemName);
+        // return existing if there is one
+        StringIO s;
+        if (userName != null) {
+            s = getByUserName(userName);
+            if (s != null) {
+                if (getBySystemName(systemName) != s) {
+                    log.error("inconsistent user ({}) and system name ({}) results; userName related to ({})", userName, systemName, s.getSystemName());
```

Comment:
I suggest throwing an `IllegalArgumentException` here.

Reply:
> I suggest throwing an IllegalArgumentException here.

This is probably a good idea, but this is what all of these managers do now: They return the existing object with the requested name, and issue that error.  We can change that behavior here, but I'm not sure I want this manager to behave like every other one in the system.  And changing all those other ones may well break scripts, etc where the user hasn't noticed any problems.